test-e2e/panels/structures.test.ts

```diff
@@ -7,36 +7,119 @@ test('should draw structure and display it with MF', async ({ page }) => {
   // Open the "Structures" panel.
   await nmrium.clickPanel('Structures');
 
-  // The SVG container should not be rendered when there are no molecules.
-  await expect(nmrium.page.locator('.mol-svg-container ')).toBeHidden();
-  // Click on the "Add Molecule" button.
-  await nmrium.page.click('data-test-id=panel-structures-button-add >> nth=0');
-
-  // Select the "aromatic ring" tool.
-  await nmrium.page.click('canvas >> nth=0', {
-    position: {
-      x: 35,
-      y: 200,
-    },
+  await test.step('empty panel', async () => {
+    // The SVG container should not be rendered when there are no molecules.
+    await expect(nmrium.page.locator('.mol-svg-container ')).toBeHidden();
   });
-  // Draw the aromatic ring.
-  await nmrium.page.click('canvas >> nth=1', {
-    position: {
-      x: 50,
-      y: 50,
-    },
+  await test.step('Add aromatic ring molecule', async () => {
+    // Click on the "Add Molecule" button.
+    await nmrium.page.click(
+      'data-test-id=panel-structures-button-add >> nth=0',
+    );
+
+    // Select the "aromatic ring" tool.
+    await nmrium.page.click('canvas >> nth=0', {
+      position: {
+        x: 35,
+        y: 200,
+      },
+    });
+    // Draw the aromatic ring.
+    await nmrium.page.click('canvas >> nth=1', {
+      position: {
+        x: 50,
+        y: 50,
+      },
+    });
+    // Save the molecule.
+    await nmrium.page.click('text=Save');
+  });
+
+  await test.step('Check the visibly of the aromatic ring molecule', async () => {
+    // The molecule SVG rendering should now be visible in the panel.
+    await expect(
+      nmrium.page.locator('.mol-svg-container #molSVG0'),
+    ).toHaveAttribute('xmlns', 'http://www.w3.org/2000/svg');
+    await expect(
+      nmrium.page.locator('.mol-svg-container #molSVG0'),
+    ).toBeVisible();
+
+    // The molecular formula should now be visible in the panel.
+    await expect(
+      nmrium.page.locator('text=C6H6 - 78.11 >> nth=0'),
+    ).toBeVisible();
+
+    // The number of molecules should now be visible in the panel.
+    await expect(nmrium.page.locator('text=1 / 1 >> nth=0')).toBeVisible();
   });
-  // Save the molecule.
-  await nmrium.page.click('text=Save');
 
-  // The molecule SVG rendering should now be visible in the panel.
-  await expect(
-    nmrium.page.locator('.mol-svg-container #molSVG0'),
-  ).toHaveAttribute('xmlns', 'http://www.w3.org/2000/svg');
+  await test.step('Add a second molecule and check the visibility', async () => {
+    // Click on the "Add Molecule" button.
+    await nmrium.page.click(
+      'data-test-id=panel-structures-button-add >> nth=0',
```

Comment:
```suggestion
      'data-test-id=panel-structures-button-add',
```

Please remove all other `nth=0` that are not necessary.